client/blockchain-service/src/handler.rs

```diff
@@ -104,6 +126,12 @@ pub struct BlockchainService {
     /// A list of Provider IDs that this node has to pay attention to submit proofs for.
     /// This could be a BSP or a list of buckets that an MSP has.
     provider_ids: Vec<ProviderId>,
+    /// A lock to prevent multiple tasks from writing to the runtime forest root (send transactions) at the same time.
+    forest_root_write_lock: Option<tokio::sync::oneshot::Receiver<()>>,
```

Comment:
*(note from author)*

This is a oneshot channel instead of a regular mutex because we want to "lock" in 1 thread (blockchain service) and unlock it at the end of the spawned task. The alternative would be to send a `MutexGuard` which in my opinion is even more confusing.

Reply:
Probably worth commenting in the code.